lib/buffer.js

```diff
@@ -1112,14 +1112,18 @@ function adjustOffset(offset, length) {
   return NumberIsNaN(offset) ? 0 : length;
 }
 
-Buffer.prototype.slice = function slice(start, end) {
+Buffer.prototype.subarray = function subarray(start, end) {
   const srcLength = this.length;
   start = adjustOffset(start, srcLength);
   end = end !== undefined ? adjustOffset(end, srcLength) : srcLength;
   const newLength = end > start ? end - start : 0;
   return new FastBuffer(this.buffer, this.byteOffset + start, newLength);
 };
 
+Buffer.prototype.slice = function slice(start, end) {
+  return this.subarray(start, end);
+};
```

Comment:
Can't we reuse the same method?
```suggestion
Buffer.prototype.slice = Buffer.prototype.subarray;
```

Reply:
I did it this way to preserve the function name (and to a lesser extent for the possibility in the future `subarray` gets another parameter and to preserve reference inequality). I don't feel strongly about it and we can set it to the same reference directly if people feel that's better.

Reply:
For built-in objects, the ES spec usually reuse the same methods and therefore the name not always match (e.g.: `Array.prototype[Symbol.iterator].name === 'values'`, or `Set.prototype.keys.name === 'values'`), imo it would make sense to do the same thing here – but I don't feel strongly either way.

Reply:
Hmm, actually I see that:

```js
Array.prototype.values === Array.prototype[Symbol.iterator]; // true in v8
```

And the spec literally says:

> The initial value of the @@iterator property is the same function object as the initial value of the Array.prototype.values property.

Neat, I didn't know the spec did that.